Licensing_Plugin_Admin.php

```diff
@@ -86,24 +95,33 @@ public function admin_head_licensing_upgrade() {
 	 */
 	function possible_state_change( $config, $old_config ) {
 		$changed = false;
+		$new_key = $config->get_string( 'plugin.license_key' );
+		$old_key = $old_config->get_string( 'plugin.license_key' );
 
-		if ( $old_config->get_string( 'plugin.license_key' ) !== '' && $config->get_string( 'plugin.license_key' ) === '' ) {
-			$result = Licensing_Core::deactivate_license( $old_config->get_string( 'plugin.license_key' ) );
+		if ( '' === $new_key && '' === $old_key ) {
+			// No new key or old key. Do nothing.
+			return;
+		} elseif ( '' === $new_key && '' !== $old_key ) {
+			// Current key set but new is blank, deactivating old.
+			$result = Licensing_Core::deactivate_license( $old_key );
 			if ( $result ) {
 				$this->site_inactivated = true;
 			}
 			$changed = true;
-		} elseif ( $old_config->get_string( 'plugin.license_key' ) === '' && $config->get_string( 'plugin.license_key' ) !== '' ) {
-			$result = Licensing_Core::activate_license( $config->get_string( 'plugin.license_key' ), W3TC_VERSION );
+		} elseif ( '' !== $new_key && $old_key === '' ) {
+			// Current key is blank but new is not, activating new.
+			$result = Licensing_Core::activate_license( $new_key, W3TC_VERSION );
 			if ( $result ) {
 				$this->site_activated = true;
 				$config->set( 'common.track_usage', true );
 			}
 			$changed = true;
-		} elseif ( $old_config->get_string( 'plugin.license_key' ) !== $config->get_string( 'plugin.license_key' ) ) {
-			$result = Licensing_Core::activate_license( $config->get_string( 'plugin.license_key' ), W3TC_VERSION );
-			if ( $result ) {
-				$this->site_activated = true;
+		} elseif ( '' !== $new_key && '' !== $old_key && $new_key !== $old_key ) {
+			// Current key is set and new different key provided. Deactivating old and activating new.
+			$deactivate_result = Licensing_Core::deactivate_license( $old_key );
+			$activate_result   = Licensing_Core::activate_license( $new_key, W3TC_VERSION );
+			if ( $deactivate_result && $activate_result ) {
```

Comment:
`Licensing_Core::deactivate_license()` returns a boolean, but `Licensing_Core::activate_license()` does not -- it returns (on success):
```
(object) array(
   'license' => 'site_active',
   'license_status' => 'active.by_rooturi',
)
```

When a key is swapped, it doesn't show the correct messages.  Each action should show a message.  For example, messages like "Your old Pro license key is not valid, so no action was taken." and "YYour Pro license key is correct and has been applied.".

Reply:
This was due to a missed reference to the old $result being missed. It should now display a message